### DISCUSSION TITLE
Question about OpenRadioss capability

### DISCUSSION THREAD

**Discussion by Majid-MHZ** [OpenRadioss/OpenRadioss#2764] [Q&A]
Dear @MarianBulla ,
I hope you are doing well.

Thank you for developing the excellent software "OpenRadioss." I have a question for you. Currently, I am simulating a powder compaction forming model using LS-PrePost software and running it with OpenRadioss. Soon, I will need to use a CP (crystal plasticity) model in my current simulation. My concern is whether OpenRadioss supports any user subroutine and can handle such a task. I would be very grateful if you could share your experience and knowledge regarding this matter with me.

Best regards,
Majid

**Comment by MarianBulla** [comment]
Dear @Majid-MHZ,
thanks for you interest in OpenRadioss. 
Regarding your question, yes sure I will help you. Can you describe more precisally your point, please ?
Do you have any papers or more in detail description on what is needed for your task ?

Thanks and best regards,
Marian

**Comment by Majid-MHZ** [comment]
Dear @MarianBulla,

Thank you for your support and kindness. We have done the powder-forming project using Abaqus software and employed a CP (crystal plasticity) model to define our desired material properties. Now, I need to replicate the procedure using open-source software. For this, I am using LS-PrePost to create my model and OpenRadioss to run the simulation.

I would like to know how I can convert the CP model used in Abaqus to a compatible format for OpenRadioss. Is there any documentation or tutorial available that explains the conversion process? What would the procedure involve?

Thank you in advance for your assistance.

Best wishes,
Majid

**Reply by MarianBulla** [reply]
Dear @Majid-MHZ,
if you are using VUMAT (in Abaqus explicite), then it is quite easy to convert the user subroutines.
From my experience, the biggest difference is in the parameter read-in procedure.
In OpenRadioss, we use a two-step approach: The STARTER and the ENGINE.
The parameter for your material model are read-in in the STARTER process.
You have the do it in the starter-routine, and pass them to the engine, where the are used:

![image](https://github.com/OpenRadioss/OpenRadioss/assets/105412502/adfab6d8-4c7c-4cbf-b001-36fc4c37e960)

(Here is just an example of another material model)

Hope it helps you ?

Thanks and best regards,
Marian

**Comment by Majid-MHZ** [comment]
Dear @MarianBulla,

Surely, it is helpful. I have a request for you. Could you please do me a favor and send me an example of a prepared user subroutine file so I can figure out how the template should be? or could you please suggest some examples or documentation to me?

Best regards,
Majid

**Reply by MarianBulla** [reply]
Dear Majid,

sure. Please find attached the famous Mooney-Rivlin material model as a USER subroutine.
(just remove the ".txt" at the end of the files)

[LUSER01.f.txt](https://github.com/user-attachments/files/16008530/LUSER01.f.txt)

[LECMUSER01.f.txt](https://github.com/user-attachments/files/16008527/LECMUSER01.f.txt)

For further documentation, please review the help/manual pages for USER Subroutines:

https://help.altair.com/hwsolvers/rad/index.htm

Hope it helps you ?

Thanks and best regards,
Marian

**Comment by Majid-MHZ** [comment]
Dear @MarianBulla 

Thank you for your help. I will work on it and will contact you if I have any questions.

Best regards,
Majid
